### PR TITLE
Add radiusSearch to KDTree

### LINKED ISSUE #12747: Add radiusSearch to KDTree
## Rationale
Use the `KDTree` proxy object to call the already implemented `radiusSearch`
capability in `nanoflann`.

## Description
Simple change by adding a function to `KDTree` and some testing.

## Impact
Additional capability.

### PR DESCRIPTION
**Pull request by snschune**
closes #12747

### CODE REVIEW

File: framework/src/utils/KDTree.C [hunk 1]
```diff
@@ -48,3 +48,26 @@ KDTree::neighborSearch(Point & query_point,
...
   return_dist_sqr.resize(n_result);
 }
+
+void
+KDTree::radiusSearch(Point & query_point,
+                     Real radius,
+                     std::vector<std::size_t> & return_index,
+                     std::vector<Real> & return_dist_sqr)
+{
+  std::vector<std::pair<std::size_t, Real>> indices_dist;
```
Comment (dschwen) [inline framework/src/utils/KDTree.C]: Any reason you don't make the signature 
```
KDTree::radiusSearch(Point & query_point,
                     Real radius,
                     std::vector<std::pair<std::size_t, Real>> indices_dist);
```
and save the copy operation?

Comment (dschwen) [reply framework/src/utils/KDTree.C]: Oh, because the other function make the same mistake... ;-)

Comment (snschune) [reply framework/src/utils/KDTree.C]: `neighborSearch` and `radiusSearch` can be changed to do it right....

Comment (snschune) [reply framework/src/utils/KDTree.C]: The inconsistency comes from the `nanoflann` interface. `radiusSearch` wants a `std::vector<std::pair<unsigned int, Real>>`, while `knnSearch` wants two separate vectors.

**Review by dschwen — APPROVED** [review]

**Pull request merged**